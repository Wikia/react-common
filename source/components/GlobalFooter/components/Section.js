import React from 'react';
import { withTranslation } from 'react-i18next';

import LinkImage from './LinkImage';
import LinkBranded from './LinkBranded';
import LinkText from './LinkText';

const renderLink = (type, model) => {
    switch (type) {
        case 'link-image':
            return <LinkImage model={model} />;
        case 'link-branded':
            return <LinkBranded model={model} />;
        default:
            return <LinkText model={model}/>;
    }
};

const Section = ({ t, model, name, parentName }) => {
    const title = model.header ? model.header.title : '';
    const description = model.description;
    const links = model.links;

    return <section className={`wds-global-footer__${parentName}-section wds-is-${name}`}>
        {title && <h3 className="wds-global-footer__section-header">{t(title.key)}</h3>}
        {description && <span className="wds-global-footer__section-description">{t(description.key)}</span>}

        <ul className="wds-global-footer__links-list">
            {links.map((link, index) =>
                <li className="wds-global-footer__links-list-item" key={index}>
                    {renderLink(link.type, link)}
                </li>
            )}

        </ul>
    </section>
};

export default withTranslation()(Section);
