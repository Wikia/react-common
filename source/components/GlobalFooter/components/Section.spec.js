import React from 'react';
import renderer from 'react-test-renderer';

import Section from './Section';

test('Section with links only renders correctly', () => {
    const component = renderer.create(
        <Section
            name="site_overview"
            model={{
                links: [
                    {
                        type: 'link-text',
                        title: {
                            type: 'translatable-text',
                            key: 'Terms of Use',
                        },
                        href: 'http://www.wikia.com/Terms_of_use',
                    },
                    {
                        type: 'link-text',
                        title: {
                            type: 'translatable-text',
                            key: 'Privacy Policy',
                        },
                        href: 'http://www.wikia.com/Privacy_Policy',
                    },
                    {
                        type: 'link-text',
                        title: {
                            type: 'translatable-text',
                            key: 'Global Sitemap',
                        },
                        href: 'http://www.wikia.com/Sitemap',
                    },
                    {
                        type: 'link-text',
                        title: {
                            type: 'translatable-text',
                            key: 'Local Sitemap',
                        },
                        href: '/wiki/Local_Sitemap',
                    },
                ],
            }}
        />,
    );

    expect(component.toJSON()).toMatchSnapshot();
});

test('Section with header renders correctly', () => {
    const component = renderer.create(
        <Section
            name="site_overview"
            model={{
                header: {
                    type: 'line-text',
                    title: {
                        type: 'translatable-text',
                        key: 'Overview',
                    },
                },
                links: [
                    {
                        type: 'link-text',
                        title: {
                            type: 'translatable-text',
                            key: 'About',
                        },
                        href: 'http://www.wikia.com/about',
                    },
                    {
                        type: 'link-text',
                        title: {
                            type: 'translatable-text',
                            key: 'Careers',
                        },
                        href: 'https://careers.wikia.com',
                    },
                    {
                        type: 'link-text',
                        title: {
                            type: 'translatable-text',
                            key: 'Press',
                        },
                        href: 'http://fandom.wikia.com/press',
                    },
                    {
                        type: 'link-text',
                        title: {
                            type: 'translatable-text',
                            key: 'Contact',
                        },
                        href: 'http://fandom.wikia.com/contact',
                    },
                ],
            }}
        />,
    );

    expect(component.toJSON()).toMatchSnapshot();
});

test('Section with description renders correctly', () => {
    const component = renderer.create(
        <Section
            name="site_overview"
            model={{
                description: {
                    type: 'translatable-text',
                    key: "Can't find a community you love? Create your own and start something epic.",
                },
                links: [
                    {
                        type: 'link-text',
                        title: {
                            type: 'translatable-text',
                            key: 'Start a wiki',
                        },
                        href: 'http://www.wikia.com/Special:CreateNewWiki',
                    },
                ],
            }}
        />,
    );

    expect(component.toJSON()).toMatchSnapshot();
});

test('Section with link-image renders correctly', () => {
    const component = renderer.create(
        <Section
            name="site_overview"
            model={{
                links: [
                    {
                        type: 'link-image',
                        image: 'wds-icons-facebook',
                        'image-data': {
                            type: 'wds-svg',
                            name: 'wds-icons-facebook',
                        },
                        title: {
                            type: 'translatable-text',
                            key: 'global-footer-follow-us-link-facebook',
                        },
                        href: '#',
                        tracking_label: 'follow-us.facebook',
                    },
                ],
            }}
        />,
    );

    expect(component.toJSON()).toMatchSnapshot();
});

test('Section with link-branded renders correctly', () => {
    const component = renderer.create(
        <Section
            name="site_overview"
            model={{
                links: [
                    {
                        type: 'link-branded',
                        brand: 'games',
                        title: {
                            type: 'translatable-text',
                            key: 'Games',
                        },
                        href: 'http://fandom.wikia.com/games',
                    },
                ],
            }}
        />,
    );

    expect(component.toJSON()).toMatchSnapshot();
});
