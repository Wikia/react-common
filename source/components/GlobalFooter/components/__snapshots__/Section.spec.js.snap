// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section with description renders correctly 1`] = `
<section
  className="wds-global-footer__wikia-section wds-is-site_overview"
>
  
  <span
    className="wds-global-footer__section-description"
  >
    Can't find a community you love? Create your own and start something epic.
  </span>
  <ul
    className="wds-global-footer__links-list"
  >
    <li
      className="wds-global-footer__links-list-item"
    >
      <a
        className="wds-global-footer__link"
        href="http://www.wikia.com/Special:CreateNewWiki"
      >
        Start a wiki
      </a>
    </li>
  </ul>
</section>
`;

exports[`Section with header renders correctly 1`] = `
<section
  className="wds-global-footer__wikia-section wds-is-site_overview"
>
  <h3
    className="wds-global-footer__section-header"
  >
    Overview
  </h3>
  <ul
    className="wds-global-footer__links-list"
  >
    <li
      className="wds-global-footer__links-list-item"
    >
      <a
        className="wds-global-footer__link"
        href="http://www.wikia.com/about"
      >
        About
      </a>
    </li>
    <li
      className="wds-global-footer__links-list-item"
    >
      <a
        className="wds-global-footer__link"
        href="https://careers.wikia.com"
      >
        Careers
      </a>
    </li>
    <li
      className="wds-global-footer__links-list-item"
    >
      <a
        className="wds-global-footer__link"
        href="http://fandom.wikia.com/press"
      >
        Press
      </a>
    </li>
    <li
      className="wds-global-footer__links-list-item"
    >
      <a
        className="wds-global-footer__link"
        href="http://fandom.wikia.com/contact"
      >
        Contact
      </a>
    </li>
  </ul>
</section>
`;

exports[`Section with link-branded renders correctly 1`] = `
<section
  className="wds-global-footer__wikia-section wds-is-site_overview"
>
  
  <ul
    className="wds-global-footer__links-list"
  >
    <li
      className="wds-global-footer__links-list-item"
    >
      <a
        className="wds-global-footer__link"
        href="http://fandom.wikia.com/games"
      >
        <div>
          Games
        </div>
      </a>
    </li>
  </ul>
</section>
`;

exports[`Section with link-image renders correctly 1`] = `
<section
  className="wds-global-footer__wikia-section wds-is-site_overview"
>
  
  <ul
    className="wds-global-footer__links-list"
  >
    <li
      className="wds-global-footer__links-list-item"
    >
      <a
        className="wds-global-footer__link"
        data-tracking-label="follow-us.facebook"
        href="#"
      >
        <svg
          className="wds-global-footer__image wds-icon"
        >
          <use
            xlinkHref="#wds-icons-facebook"
          />
        </svg>
      </a>
    </li>
  </ul>
</section>
`;

exports[`Section with links only renders correctly 1`] = `
<section
  className="wds-global-footer__wikia-section wds-is-site_overview"
>
  
  <ul
    className="wds-global-footer__links-list"
  >
    <li
      className="wds-global-footer__links-list-item"
    >
      <a
        className="wds-global-footer__link"
        href="http://www.wikia.com/Terms_of_use"
      >
        Terms of Use
      </a>
    </li>
    <li
      className="wds-global-footer__links-list-item"
    >
      <a
        className="wds-global-footer__link"
        href="http://www.wikia.com/Privacy_Policy"
      >
        Privacy Policy
      </a>
    </li>
    <li
      className="wds-global-footer__links-list-item"
    >
      <a
        className="wds-global-footer__link"
        href="http://www.wikia.com/Sitemap"
      >
        Global Sitemap
      </a>
    </li>
    <li
      className="wds-global-footer__links-list-item"
    >
      <a
        className="wds-global-footer__link"
        href="/wiki/Local_Sitemap"
      >
        Local Sitemap
      </a>
    </li>
  </ul>
</section>
`;
